File: PRACTICE FOLDER/RECURSION/gcd.py
"""

"""

from typing import List, Optional

class Solution:
    def gcd(self, a, b):
        if b==0:
        return a
        return gcd(b,a%b)


if __name__ == "__main__":
    sol = Solution()
    a =48,b=98
    print(f"GCD OF {a}")

File: PRACTICE FOLDER/RECURSION/gcd.cpp
#include<iostream>
using namespace std;

class Solution{
    public:
    int gcd(int a,int b){
        if(b==0){
            return a;
        }
        return gcd(b,a%b);
    }

};

int main(){
    Solution sol;
    int a=48,b=98;
    cout<<"GCD OF "<<a<<" and "<<b<<"  is: "<<sol.gcd(a,b)<<endl;
    return 0;
}